File: src/source-language-catalog.js
'use strict';

const CUTLETS = Object.freeze([
  Object.freeze({ canonicalIndex: 1, text: 'bronze' }),
  Object.freeze({ canonicalIndex: 2, text: 'vulpe' }),
  Object.freeze({ canonicalIndex: 3, text: 'ren' }),
  Object.freeze({ canonicalIndex: 4, text: 'larice' }),
  Object.freeze({ canonicalIndex: 5, text: 'pense' }),
  Object.freeze({ canonicalIndex: 6, text: 'quar partes de nin' }),
  Object.freeze({ canonicalIndex: 7, text: 'Palgursh' }),
  Object.freeze({ canonicalIndex: 8, text: 'papirus' }),
  Object.freeze({ canonicalIndex: 9, text: 'grappe' }),
  Object.freeze({ canonicalIndex: 10, text: 'scorpion' }),
  Object.freeze({ canonicalIndex: 11, text: 'cindre' }),
  Object.freeze({ canonicalIndex: 12, text: 'frument' }),
  Object.freeze({ canonicalIndex: 13, text: 'fluvie' }),
  Object.freeze({ canonicalIndex: 14, text: 'rise' }),
  Object.freeze({ canonicalIndex: 15, text: 'Akkad' }),
  Object.freeze({ canonicalIndex: 16, text: 'corn' }),
  Object.freeze({ canonicalIndex: 17, text: 'li vacui vase' })
]);

const MONTHS = Object.freeze([
  Object.freeze({ canonicalIndex: 1, text: 'argile' }),
  Object.freeze({ canonicalIndex: 2, text: 'granat' }),
  Object.freeze({ canonicalIndex: 3, text: 'cubit' }),
  Object.freeze({ canonicalIndex: 4, text: 'invidie' }),
  Object.freeze({ canonicalIndex: 5, text: 'Eridu' }),
  Object.freeze({ canonicalIndex: 6, text: 'dent-pasta' }),
  Object.freeze({ canonicalIndex: 7, text: 'tri partes de quin' }),
  Object.freeze({ canonicalIndex: 8, text: 'Karshumb' }),
  Object.freeze({ canonicalIndex: 9, text: 'leopard' }),
  Object.freeze({ canonicalIndex: 10, text: 'stann' }),
  Object.freeze({ canonicalIndex: 11, text: 'brume' }),
  Object.freeze({ canonicalIndex: 12, text: 'oliban' }),
  Object.freeze({ canonicalIndex: 13, text: 'fus' }),
  Object.freeze({ canonicalIndex: 14, text: 'costa' }),
  Object.freeze({ canonicalIndex: 15, text: 'carob' }),
  Object.freeze({ canonicalIndex: 16, text: 'Uruk' }),
  Object.freeze({ canonicalIndex: 17, text: 'honte' }),
  Object.freeze({ canonicalIndex: 18, text: 'camel' }),
  Object.freeze({ canonicalIndex: 19, text: 'cupr' }),
  Object.freeze({ canonicalIndex: 20, text: 'pute' }),
  Object.freeze({ canonicalIndex: 21, text: 'vitelle' }),
  Object.freeze({ canonicalIndex: 22, text: 'stelle' }),
  Object.freeze({ canonicalIndex: 23, text: 'mel' }),
  Object.freeze({ canonicalIndex: 24, text: 'splen' }),
  Object.freeze({ canonicalIndex: 25, text: 'calcari' }),
  Object.freeze({ canonicalIndex: 26, text: 'joy' }),
  Object.freeze({ canonicalIndex: 27, text: 'fig' }),
  Object.freeze({ canonicalIndex: 28, text: 'Ninive' }),
  Object.freeze({ canonicalIndex: 29, text: 'ran' }),
  Object.freeze({ canonicalIndex: 30, text: 'gudron' }),
  Object.freeze({ canonicalIndex: 31, text: 'candel' }),
  Object.freeze({ canonicalIndex: 32, text: 'li cludet porta' }),
  Object.freeze({ canonicalIndex: 33, text: 'sesam' }),
  Object.freeze({ canonicalIndex: 34, text: 'nuca' }),
  Object.freeze({ canonicalIndex: 35, text: 'argent' }),
  Object.freeze({ canonicalIndex: 36, text: 'lilie' }),
  Object.freeze({ canonicalIndex: 37, text: 'tempeste' }),
  Object.freeze({ canonicalIndex: 38, text: 'asin' }),
  Object.freeze({ canonicalIndex: 39, text: 'farine' }),
  Object.freeze({ canonicalIndex: 40, text: 'regret' }),
  Object.freeze({ canonicalIndex: 41, text: 'Babylon' }),
  Object.freeze({ canonicalIndex: 42, text: 'lingue' }),
  Object.freeze({ canonicalIndex: 43, text: 'lin' }),
  Object.freeze({ canonicalIndex: 44, text: 'sal' }),
  Object.freeze({ canonicalIndex: 45, text: 'pir' }),
  Object.freeze({ canonicalIndex: 46, text: 'arc' }),
  Object.freeze({ canonicalIndex: 47, text: 'sand' })
]);

const SourceLanguageCatalog = Object.freeze({
  version: '1.0.0-stage-01',
  language: 'Interlingue / Occidental',
  cutlets: CUTLETS,
  months: MONTHS
});

function textByCanonicalIndex(group, canonicalIndex) {
  const rows = group === 'cutlet' ? SourceLanguageCatalog.cutlets : group === 'month' ? SourceLanguageCatalog.months : null;
  if (rows === null || !Number.isInteger(canonicalIndex) || canonicalIndex < 1 || canonicalIndex > rows.length) {
    throw new RangeError('Índice canonic ínvalid in li catalog de lingue-fonte.');
  }
  return rows[canonicalIndex - 1].text;
}

module.exports = Object.freeze({ SourceLanguageCatalog, textByCanonicalIndex });

File: lib/src/source_language_catalog.dart
final class CanonicalName {
  final int canonicalIndex;
  final String text;

  const CanonicalName(this.canonicalIndex, this.text);
}

final class SourceLanguageCatalog {
  static const String version = '1.0.0';
  static const String naturalLanguage = 'עברית';

  static const List<CanonicalName> cutletNames = <CanonicalName>[
    CanonicalName(1, 'ארד'),
    CanonicalName(2, 'שועל'),
    CanonicalName(3, 'כליה'),
    CanonicalName(4, 'לגש'),
    CanonicalName(5, 'מחשבה'),
    CanonicalName(6, 'ארבעה חלקים מתשעה'),
    CanonicalName(7, 'פַּלְגּוּרַשׁ'),
    CanonicalName(8, 'גומא'),
    CanonicalName(9, 'אשכול'),
    CanonicalName(10, 'עקרב'),
    CanonicalName(11, 'אפר'),
    CanonicalName(12, 'חיטה'),
    CanonicalName(13, 'נהר'),
    CanonicalName(14, 'צחוק'),
    CanonicalName(15, 'אכד'),
    CanonicalName(16, 'קרן'),
    CanonicalName(17, 'הכד הריק'),
  ];

  static const List<CanonicalName> monthNames = <CanonicalName>[
    CanonicalName(1, 'טין'),
    CanonicalName(2, 'רימון'),
    CanonicalName(3, 'מרפק'),
    CanonicalName(4, 'קנאה'),
    CanonicalName(5, 'ארידו'),
    CanonicalName(6, 'משחת־שיניים'),
    CanonicalName(7, 'שלושה חלקים מחמישה'),
    CanonicalName(8, 'כַּרְשׁוּמַב'),
    CanonicalName(9, 'נמר'),
    CanonicalName(10, 'בדיל'),
    CanonicalName(11, 'ערפל'),
    CanonicalName(12, 'לבונה'),
    CanonicalName(13, 'כישור'),
    CanonicalName(14, 'צלע'),
    CanonicalName(15, 'חרוב'),
    CanonicalName(16, 'אורוק'),
    CanonicalName(17, 'בושה'),
    CanonicalName(18, 'גמל'),
    CanonicalName(19, 'נחושת'),
    CanonicalName(20, 'באר'),
    CanonicalName(21, 'חלמון'),
    CanonicalName(22, 'כוכב'),
    CanonicalName(23, 'דבש'),
    CanonicalName(24, 'טחול'),
    CanonicalName(25, 'אבן־גיר'),
    CanonicalName(26, 'שמחה'),
    CanonicalName(27, 'תאנה'),
    CanonicalName(28, 'נינוה'),
    CanonicalName(29, 'צפרדע'),
    CanonicalName(30, 'זפת'),
    CanonicalName(31, 'נר'),
    CanonicalName(32, 'הדלת הסגורה'),
    CanonicalName(33, 'שומשום'),
    CanonicalName(34, 'עורף'),
    CanonicalName(35, 'כסף'),
    CanonicalName(36, 'שושן'),
    CanonicalName(37, 'סערה'),
    CanonicalName(38, 'חמור'),
    CanonicalName(39, 'קמח'),
    CanonicalName(40, 'חרטה'),
    CanonicalName(41, 'בבל'),
    CanonicalName(42, 'לשון'),
    CanonicalName(43, 'פשתן'),
    CanonicalName(44, 'מלח'),
    CanonicalName(45, 'אגס'),
    CanonicalName(46, 'קשת'),
    CanonicalName(47, 'חול'),
  ];

  static String cutletText(int canonicalIndex) {
    if (canonicalIndex < 1 || canonicalIndex > cutletNames.length) {
      throw RangeError.range(canonicalIndex, 1, cutletNames.length, 'canonicalIndex');
    }
    return cutletNames[canonicalIndex - 1].text;
  }

  static String monthText(int canonicalIndex) {
    if (canonicalIndex < 1 || canonicalIndex > monthNames.length) {
      throw RangeError.range(canonicalIndex, 1, monthNames.length, 'canonicalIndex');
    }
    return monthNames[canonicalIndex - 1].text;
  }
}
